diff --git a/arch/arm/mach-sunxi/dram_helpers.c b/arch/arm/mach-sunxi/dram_helpers.c
index 83dbe4c..c78fdee 100644
--- a/arch/arm/mach-sunxi/dram_helpers.c
+++ b/arch/arm/mach-sunxi/dram_helpers.c
@@ -20,8 +20,10 @@ void mctl_await_completion(u32 *reg, u32 mask, u32 val)
 	unsigned long tmo = timer_get_us() + 1000000;
 
 	while ((readl(reg) & mask) != val) {
-		if (timer_get_us() > tmo)
-			panic("Timeout initialising DRAM\n");
+		if (timer_get_us() > tmo) {
+			printf("DRAM TIMEOUT! (graceful fallback)\n");
+			return;
+		}
 	}
 }
 
diff --git a/arch/arm/mach-sunxi/dram_sun8i_a33.c b/arch/arm/mach-sunxi/dram_sun8i_a33.c
index 0d08b6a..873faba 100644
--- a/arch/arm/mach-sunxi/dram_sun8i_a33.c
+++ b/arch/arm/mach-sunxi/dram_sun8i_a33.c
@@ -228,9 +228,9 @@ static int mctl_channel_init(struct dram_para *para)
 	else
 		setbits_le32(&mctl_ctl->pllgcr, 0x3 << 18);
 
-	/* Auto detect dram config, set 2 rank and 16bit bus-width */
+	/* Use rank passed from init */
 	para->cs1 = 0;
-	para->rank = 2;
+	// para->rank = 2; 
 	para->bus_width = 16;
 	mctl_set_cr(para);
 
@@ -335,7 +335,7 @@ unsigned long sunxi_dram_init(void)
 	struct dram_para para = {
 		.cs1 = 0,
 		.bank = 1,
-		.rank = 2,
+		.rank = 1,
 		.rows = 15,
 		.bus_width = 16,
 		.page_size = 2048,
diff --git a/board/sunxi/board.c b/board/sunxi/board.c
index ac9cefc..10472fd 100644
--- a/board/sunxi/board.c
+++ b/board/sunxi/board.c
@@ -547,6 +547,20 @@ static void sunxi_spl_store_dram_size(phys_addr_t dram_size)
 	spl->dram_size = dram_size >> 20;
 }
 
+
+#ifdef CONFIG_GA36_BACKLIGHT_BEACON
+#define GA36_PIO_PH_BASE (SUNXI_PIO_BASE + SUNXI_GPIO_H * SUNXI_PINCTRL_BANK_SIZE)
+static void ga36_backlight_set_spl(int on)
+{
+	u32 val = readl(GA36_PIO_PH_BASE + 0x10);
+	if (on)
+		val |= (1U << 0);
+	else
+		val &= ~(1U << 0);
+	writel(val, GA36_PIO_PH_BASE + 0x10);
+}
+#endif
+
 void sunxi_board_init(void)
 {
 	int power_failed = 0;
@@ -567,6 +581,8 @@ void sunxi_board_init(void)
 	defined CONFIG_AXP803_POWER
 	power_failed = axp_init();
 
+
+
 	if (IS_ENABLED(CONFIG_AXP_DISABLE_BOOT_ON_POWERON) && !power_failed) {
 		u8 boot_reason;
 
@@ -633,6 +649,19 @@ void sunxi_board_init(void)
 	printf("DRAM:");
 	gd->ram_size = sunxi_dram_init();
 	printf(" %d MiB\n", (int)(gd->ram_size >> 20));
+
+#ifdef CONFIG_GA36_BACKLIGHT_BEACON
+	pmic_bus_setbits(0x12, (1U << 7));
+	clrsetbits_le32(GA36_PIO_PH_BASE, 0xf, SUNXI_GPIO_OUTPUT);
+	ga36_backlight_set_spl(1);
+	mdelay(200);
+	ga36_backlight_set_spl(0);
+	mdelay(200);
+	ga36_backlight_set_spl(1);
+	mdelay(200);
+	ga36_backlight_set_spl(0);
+#endif
+
 	if (!gd->ram_size)
 		hang();
 
